app/api/internal/applications.rb

```diff
@@ -18,17 +18,25 @@ module API
           end
           app_attrs[:service_id] = service_id
           app_attrs[:id] = id
-          app = Application.save(app_attrs)
-          [201, headers, { status: :created, application: app.to_hash }.to_json]
+          begin
+            app = Application.save(app_attrs)
+            [201, headers, { status: :created, application: app.to_hash }.to_json]
```

Comment:
minor nit: this statement will not raise `ApplicationHasNoState` - it can either go outside or in the `else` clause of the rescue:
```ruby
begin
  app = Application.save(app_attrs)
rescue ApplicationHasNoState
  # ...
else
  [201, headers, { status: :created, application: app.to_hash }.to_json]
end
```

Reply:
I've moved the response encoding to an else of the rescue clause. I've not moved it outside because in the case of an exception it would execute both the content inside the rescue and the content outside of it.